keras_cv/models/adain.py

```diff
@@ -0,0 +1,228 @@
+# Copyright 2022 The KerasCV Authors
```

Comment:
API discussion:

- we want two functions, one that creates an encoder, decoder, and functional model end to end.  return all 3
- another function that calls the first, drops the first two, and loads weights into the functional model.